app/connector/aws-ddb/src/main/java/io/syndesis/connector/aws/ddb/customizer/DDBConnectorCustomizer.java

```diff
@@ -134,25 +136,21 @@ protected void doBeforeProducer(Exchange exchange) {
                     if (entry.getKey().startsWith("#") && entry.getValue() != null) {
                         final String searchKey = ":" + entry.getKey();
                         final String replacement = entry.getValue().toString();
-                        for (Map.Entry<String, Object> option : options.entrySet()) {
-                            if (option.getValue() != null &&
-                                option.getValue() instanceof String) {
-                                final String oldValue = option.getValue().toString();
-                                final String key = option.getKey();
-                                final String newValue = oldValue.replace(searchKey, replacement);
-
-                                LOG.trace("this.option: " + key + ":" + oldValue + "->" + newValue);
-                                this.options.put(key, newValue);
-                            }
-                        }
+
+                        element = element.replace(searchKey, replacement);
+                        attributes = attributes.replace(searchKey, replacement);
                     }
                 }
             }
         }
 
-        LOG.trace("post this.options: " + this.options);
+        Map<String, Object> options = new HashMap<String, Object>();
+        options.put("element", element);
+        options.put("attributes", attributes);
+
+        LOG.trace("post this.options: " + options);
 
-        customize(exchange, this.options);
```

Comment:
Instead of using main this.options, use a customized options map for each run. This way, variable parameters are really variable.